src/main/java/mods/eln/node/IPublishable.kt

```diff
@@ -10,7 +10,8 @@ interface IPublishable {
 class published<T>(var value: T) : ReadWriteProperty<IPublishable, T> {
     override fun getValue(thisRef: IPublishable, property: KProperty<*>): T = this.value
     override fun setValue(thisRef: IPublishable, property: KProperty<*>, value: T) {
+        val changed = this.value != value
```

Comment:
This is correct... but I initially thought it was incorrect. It's worth reading https://kotlinlang.org/docs/reference/equality.html, as it's a difference from Java.

(Reference equality could give false negatives here.)